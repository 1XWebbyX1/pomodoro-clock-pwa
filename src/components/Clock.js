import React from 'react'
import '@fortawesome/fontawesome-free/css/all.css'

class Clock extends React.Component {
    render() {
      return (
        <div class="clock">
          <p id='text'>SESSION</p>
          <time id='time'>{this.props.time}</time>
          <i className='refresh fa fa-sync-alt' onClick={this.props.onClick}></i>
        </div>
      );
    }
}

export default Clock;
