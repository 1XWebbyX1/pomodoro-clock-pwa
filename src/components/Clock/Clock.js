import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSyncAlt} from '@fortawesome/free-solid-svg-icons'

const Clock = React.forwardRef((props, ref) => (
  <div className={"clock " + props.animClass} >
    <p ref={ref} id='text'>SESSION</p>
    <time id='time'>{props.time}</time>
    <FontAwesomeIcon className='refresh' icon={faSyncAlt} onClick={props.onClick}/>
  </div>
));

export default Clock;
