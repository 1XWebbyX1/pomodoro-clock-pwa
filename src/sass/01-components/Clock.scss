.clock{
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: calc(10em + 8vw);
  height: calc(10em + 8vw);
  background-color: transparent;
  border-radius: 50%;
  color: white;
  box-shadow: 0px 3px 5px rgba(244, 244, 244, 0.3) inset;
  transition: all 1s ease-in-out;

  #time{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: 4em;
    cursor: default;
    user-select: none;
  }

  #text{
    position: absolute;
    top: 10%;
    left: 50%;
    font-size: 1em;
    transform: translateX(-50%);
    cursor: default;
    user-select: none;
  }

  .refresh{
    position: absolute;
    top: 88%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: calc(0.5em + 1.5vw);
    cursor: pointer;
  }
}

.animate{
  box-shadow: 0px 3px 5px #FF6D00 inset;
  border: 4px solid #013235;
}

.stopAnimation{
  border: 4px solid #0d252d;
  box-shadow: 0px 3px 5px rgba(244, 244, 244, 0.3) inset;
}
