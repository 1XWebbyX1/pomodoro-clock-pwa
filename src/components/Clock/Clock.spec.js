// External Depedencies
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
// Our Component
import Clock from './Clock';


configure({ adapter: new Adapter() });

describe('Clock', () => {
  const onClick = jest.fn();

  it("renders correctly", () => {
     const output = renderer.create(
         <Clock time="24:00" onClick={onClick}/>
   ).toJSON();
     expect(output).to.matchSnapshot(); //this will confirm content received through props
  });

})
