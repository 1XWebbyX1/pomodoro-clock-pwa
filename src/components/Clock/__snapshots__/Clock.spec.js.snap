// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock renders correctly 1`] = `
<div
  class="clock"
>
  <p
    id="text"
  >
    SESSION
  </p>
  <time
    id="time"
  >
    24:00
  </time>
  <i
    className="refresh fa fa-sync-alt"
    onClick={[MockFunction]}
  />
</div>
`;
